numba/npyufunc/_ufunc.c

```diff
@@ -3,16 +3,15 @@
 #include "_internal.h"
 
 static int
-get_string(PyObject *obj, char **s, const char *type_error_message)
+get_string(PyObject *obj, const char **s, const char *type_error_message)
 {
-    *s = NULL;
```

Comment:
Why is it OK to remove this?

Reply:
Think it is ok because: https://docs.python.org/2/c-api/string.html#c.PyString_AsString says

> Return a NUL-terminated representation of the contents of string. The pointer refers to the internal buffer of string, not a copy. The data must not be modified in any way, unless the string was just created using PyString_FromStringAndSize(NULL, size). It must not be deallocated. If string is a Unicode object, this function computes the default encoding of string and operates on that. If string is not a string object at all, PyString_AsString() returns NULL and raises TypeError.

which I interpreted to mean, if the input is valid it will return something valid as the pointer, but if the conversion fails then it will return NULL as the pointer and raise TypeError. I'm happy to change if back if you think it is safer?

Reply:
That makes sense, although the code path where `None` is passed in for obj will return 0 and not set the pointer at all, if I'm reading this correctly.

Reply:
I think I'll just add the line back in :) it's safer.